producer/httprequest.go

```diff
@@ -78,9 +82,69 @@ func (prod *HTTPRequest) Configure(conf core.PluginConfigReader) error {
 	prod.encoding = conf.GetString("Encoding", "text/plain; charset=utf-8")
 	prod.rawPackets = conf.GetBool("RawData", true)
 
+	//prod.Log.Debug.Printf(
+	//"protocol=%s, host=%s, port=%s, address=%s, encoding=%s, rawPackets=%d",
+	//	prod.protocol, prod.host, prod.port, prod.address, prod.encoding, prod.rawPackets)
+
+	// Health check to ping the backend with an HTTP GET
+	healthcheck.AddEndpointPathArray(
```

Comment:
Also as discussed in the office.
My original idea was to implement the health check as an interface. That way it would be sufficient to implement one method and registration happens during startup.
Good:
- easy for the developer
Bad:
- Multiple endpoints would be a breaking change
- "magic" vs. type hierarchy ... although it's debatable what's really worse

Using the type hierarchy is possible as currently all plugins derive directly or indirectly from Simple*. One can choose not to but in that case you're going all in anyway. Aka - if you want all the pain, you have to deal with it.
But even here a `RegisterEndpoint(name string, callback func()result)` would work fine.
Good:
- Multiple endpoints are possible by default
Bad:
- "magic" vs. type hierarchy (see above, not sure if good or bad)

I would say, let's make a pair session on Tuesday and see how both solutions feel and select by that.